File: src/color.rs
use std::io;

use crate::vec3::Color;

pub(crate) fn write_color<O>(out: &mut O, pixel_color: &Color) -> io::Result<()>
where
    O: std::io::Write,
{
    writeln!(
        out,
        "{} {} {}",
        (255.999 * pixel_color.x()) as i32,
        (255.999 * pixel_color.y()) as i32,
        (255.999 * pixel_color.z()) as i32
    )
}

File: src/color.zig
const std = @import("std");
const Color = @import("Vec3.zig");

pub fn writeColor(out: anytype, color: Color) !void {
    try out.writeAll(&[_]u8{
        @floatToInt(u8, 255.999 * color.e[0]),
        @floatToInt(u8, 255.999 * color.e[1]),
        @floatToInt(u8, 255.999 * color.e[2]),
    });
}
